pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -83,12 +89,17 @@ public Producer(Topic topic, ServerCnx cnx, long producerId, String producerName
         this.msgIn = new Rate();
         this.isNonPersistentTopic = topic instanceof NonPersistentTopic;
         this.msgDrop = this.isNonPersistentTopic ? new Rate() : null;
+
+        this.metadata = metadata != null
+                ? Collections.unmodifiableMap(new HashMap<>(metadata)) : Collections.emptyMap();
```

Comment:
do we need `new HashMap<>(metadata)` here? could it just be `metadata`?

Reply:
I will update to only use metadata.